crates/rustc_codegen_spirv/src/builder/spirv_asm.rs

```diff
@@ -256,10 +256,47 @@ impl<'cx, 'tcx> Builder<'cx, 'tcx> {
                 self.err("OpTypeArray in asm! is not supported yet");
                 return;
             }
+            Op::TypeRuntimeArray => SpirvType::RuntimeArray {
+                element: inst.operands[0].unwrap_id_ref(),
+            }
+            .def(self.span(), self),
+            Op::TypePointer => SpirvType::Pointer {
+                storage_class: inst.operands[0].unwrap_storage_class(),
+                pointee: inst.operands[1].unwrap_id_ref(),
+            }
+            .def(self.span(), self),
+            Op::TypeImage => SpirvType::Image {
+                sampled_type: inst.operands[0].unwrap_id_ref(),
+                dim: inst.operands[1].unwrap_dim(),
+                depth: inst.operands[2].unwrap_literal_int32(),
+                arrayed: inst.operands[3].unwrap_literal_int32(),
+                multisampled: inst.operands[4].unwrap_literal_int32(),
+                sampled: inst.operands[5].unwrap_literal_int32(),
+                image_format: inst.operands[6].unwrap_image_format(),
+                access_qualifier: None,
+            }
+            .def(self.span(), self),
             Op::TypeSampledImage => SpirvType::SampledImage {
                 image_type: inst.operands[0].unwrap_id_ref(),
             }
             .def(self.span(), self),
+            Op::Extension => {
+                self.emit_global()
+                    .extension(inst.operands[0].unwrap_literal_string());
+                return;
+            }
+            Op::Capability => {
+                self.emit_global()
+                    .capability(inst.operands[0].unwrap_capability());
+                return;
+            }
```

Comment:
Why are these needed?

Reply:
My guess was that `emit_global` helped with avoiding inserting these into the current block in a function, but maybe `rspirv` handles that already?

Or, wait, should `OpExtension`/`OpCapability` never be manually specified in `asm!`, and always deduced from other instructions?

Reply:
They weren't supported before and were needed in my test sample, I can move them to a different PR if you feel the need.

@eddyb `OpCapability` and `OpExtension` cant always be deduced from the SPIR-V opcode unfortunately. In my testing I needed `SPV_EXT_descriptor_indexing` and `RuntimeDescriptorArray` to be able to index into an `OpTypeRuntimeArray` of images, which otherwise just isn't allowed.